scopus/scopus_author_search.py

```diff
@@ -0,0 +1,117 @@
+import os
+import sys
+import xml.etree.ElementTree as ET
+import json
+
+from scopus.utils import download, ns
+from scopus.scopus_api import ScopusAbstract
+
+SCOPUS_AUTHOR_SEARCH_DIR = os.path.expanduser('~/.scopus/author_search')
+
+if not os.path.exists(SCOPUS_AUTHOR_SEARCH_DIR):
+    os.makedirs(SCOPUS_AUTHOR_SEARCH_DIR)
+
+
+FIELDS = ['eid', 'preferred-name', 'affiliation-current']
+
+class ScopusAuthorSearch(object):
+    @property
+    def authors(self):
+        """List of Authors retrieved."""
+        return self._AUTHORS
+
+    def __init__(self, query, fields=FIELDS, count=200, start=0, max_entries=5000, 
+                 refresh=False):
+        """Class to search a query, and retrieve a list of author IDs as results.
+
+        Parameters
+        ----------
+        query : str
+            A string of the query. E.g. "authlast(Einstein) and authfirst(Albert)"
+        
+        fields : str (optional, default=['eid', 'preferred-name', 
+            'affiliation-current'])
+            The fields you want returned.  Allowed fields are specified in
+            https://dev.elsevier.com/guides/AuthorSearchViews.htm.  
+        
+        count : int (optional, default=200)
+            The number of entries to be displayed at once.  A smaller number
+            means more queries with each query having less results.
+
+        start : int (optional, default=0)
+            The entry number of the first search item to start with.
+
+        refresh : bool (optional, default=False)
+            Whether to refresh the cached file if it exists or not.
+            
+        max_entries : int (optional, default=5000)
+            Raise error when the number of results is beyond this number.
+            The Scopus Search Engine does not allow more than 5000 entries.
+        
+        Raises
+        ------
+        Exception
+            If the number of search results exceeds max_entries.
+
+        Notes
+        -----
+        XML results are cached in ~/.scopus/author_search/{query}.
+
+        The Authors are stored as a property named Authors.
+        """
+
+        self.query = query
+        qfile = os.path.join(SCOPUS_AUTHOR_SEARCH_DIR,
+                             # We need to remove / in a DOI here so we can save
+                             # it as a file.
+                             query.replace('/', '_slash_'))
+
```

Comment:
The problem will be addressed differently in Socpus 1.0, namely using a hexed version of the query.  See https://github.com/scopus-api/scopus/issues/6 and https://github.com/scopus-api/scopus/pull/40 if you are keen to implement this solution here already.  Otherwise files downloaded with versions lower than 1.0 can't be read.